dspy/retrievers/embeddings.py

```diff
@@ -87,3 +87,92 @@ def _rerank_and_predict(self, q_embeds: np.ndarray, candidate_indices: np.ndarra
     def _normalize(self, embeddings: np.ndarray):
         norms = np.linalg.norm(embeddings, axis=1, keepdims=True)
         return embeddings / np.maximum(norms, 1e-10)
+
+    def save(self, path: str):
+        """
+        Save the embeddings index to disk.
+
+        This saves the corpus, embeddings, FAISS index (if present), and configuration
+        to allow for fast loading without recomputing embeddings.
+
+        Args:
+            path: Directory path where the embeddings will be saved
+        """
+        os.makedirs(path, exist_ok=True)
+
+        # Save configuration and corpus
+        config = {
+            "k": self.k,
+            "normalize": self.normalize,
+            "corpus": self.corpus,
+            "has_faiss_index": self.index is not None,
+        }
+
+        with open(os.path.join(path, "config.pkl"), "wb") as f:
+            pickle.dump(config, f)
+
+        # Save embeddings
+        np.save(os.path.join(path, "corpus_embeddings.npy"), self.corpus_embeddings)
+
+        # Save FAISS index if it exists
+        if self.index is not None:
+            try:
+                import faiss
+                faiss.write_index(self.index, os.path.join(path, "faiss_index.bin"))
+            except ImportError:
+                # If FAISS is not available, we can't save the index
+                # but we can still save the embeddings for brute force search
+                pass
+
+    def load(self, path: str, embedder):
```

Comment:
How do we want this `load` to be called? Do we want users to first create an `Embedding` instance then call `embedding.load()`, or make it a class method that return a loaded embedding?

Reply:
This is actually only for consistency with other APIs like module.load. I guess mostly people will use from_saved. Shall we actually make load a classmethod?

Reply:
This was necessary because from_saved bypasses `__init__` and self.load is called inside `from_saved`. Let me move this logic to `from_saved` then.